src/core/server/core_usage_stats/core_usage_stats_client.ts

```diff
@@ -0,0 +1,163 @@
+  ) {}
+
+  public async getUsageStats() {
+    this.debugLogger('getUsageStats() called');
+    let coreUsageStats: CoreUsageStats = {};
+    try {
+      const result = await this.repository.get<CoreUsageStats>(
+        CORE_USAGE_STATS_TYPE,
+        CORE_USAGE_STATS_TYPE
+      );
+      coreUsageStats = result.attributes;
+    } catch (err) {
+      // do nothing
+    }
+    return coreUsageStats;
+  }
+
+  public async incrementSavedObjectsImport({
+    createNewCopies,
+    overwrite,
+  }: IncrementSavedObjectsImportOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const { savedObjectsImport: current = SAVED_OBJECTS_IMPORT_DEFAULT } = apiCalls;
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsImport: {
+          total: current.total + 1,
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+          overwriteEnabled: incrementBooleanCounter(current.overwriteEnabled, overwrite),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  public async incrementSavedObjectsResolveImportErrors({
+    createNewCopies,
+  }: IncrementSavedObjectsResolveImportErrorsOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const {
+      savedObjectsResolveImportErrors: current = SAVED_OBJECTS_RESOLVE_IMPORT_ERRORS_DEFAULT,
+    } = apiCalls;
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsResolveImportErrors: {
+          total: current.total + 1,
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  public async incrementSavedObjectsExport({
+    types,
+    supportedTypes,
+  }: IncrementSavedObjectsExportOptions) {
+    const coreUsageStats = await this.getUsageStats();
+    const { apiCalls = {} } = coreUsageStats;
+    const { savedObjectsExport: current = SAVED_OBJECTS_EXPORT_DEFAULT } = apiCalls;
+    const isAllTypesSelected = !!types && supportedTypes.every((x) => types.includes(x));
+
+    const attributes = {
+      ...coreUsageStats,
+      apiCalls: {
+        ...apiCalls,
+        savedObjectsExport: {
+          total: current.total + 1,
+          allTypesSelected: incrementBooleanCounter(current.allTypesSelected, isAllTypesSelected),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  private async updateUsageStats(attributes: CoreUsageStats) {
+    const options = { id: CORE_USAGE_STATS_TYPE, overwrite: true };
+    return this.repository.create(CORE_USAGE_STATS_TYPE, attributes, options);
```

Comment:
by overwriting we're potentially loosing "counts" from other nodes. For usage data this isn't the end of the world, but we do have `SavedObjectsRepository.incrementCounter` which I think would be more elegant than reading/write.

Reply:
This is intended. I discussed this a bit in this comment: https://github.com/elastic/kibana/pull/83575#discussion_r527209893

Unfortunately `SavedObjectsRepository.incrementCounter` only allows us to increment one field at a time, so we can't use that. And as I mentioned above, this situation is pretty low-risk, as these APIs aren't hot code paths and the impact is quite minimal.

Reply:
I've opened https://github.com/elastic/kibana/pull/84326 to change the api so that it accepts several counters because this is something that will be generally useful for other plugins. Adopting `incrementCounter` will change how the stats are persisted because incrementCounter doesn't support nesting so the entire stats object will look something like:
```
interface CoreUsageStats {
  'apiCalls.savedObjectsImport.total': number;
  'apiCalls.savedObjectsImport.kibanaRequest.yes': number;
  'apiCalls.savedObjectsImport.kibanaRequest.no': number;
}
```

Since Kibana anyway flattens all fields for visualization/searching I don't think this has any impact for our telemetry.

Reply:
OK, now that that's merged I've updated the usage stats client to use incrementCounter instead: 60c3820